### PR TITLE
Reorder helper functions in fonts.py, clarify 'times' vs 'times new roman', and clarify 'usepackage{times}' vs 'usepackage{ptm}'

### PR DESCRIPTION
**Pull request by libbylbaker**
This PR tries to make the font file more transparent. 

- The functions have been reordered so the helper functions, e.g. font families are at the bottom, and paper font functions are at the top
- Font selection functions have been renamed to make clear which fonts they're defining 
- Neurips and ICML paper use times (instead of Times New Roman), so this has been corrected

### CODE REVIEW

File: tueplots/fonts.py [hunk 1]
```diff
@@ -3,147 +3,181 @@
...
+#     return {
+#         "text.usetex": False,
+#         "font.serif": ["Times New Roman"],
+#         "mathtext.fontset": "stix",  # free ptmx replacement, for ICML and NeurIPS
+#         "mathtext.rm": "Times New Roman",
+#         "mathtext.it": "Times New Roman:italic",
+#         "mathtext.bf": "Times New Roman:bold",
+#         "font.family": family,
+#     }
+
```
Comment (pnkraemer) [inline tueplots/fonts.py:173]: Let's delete this; otherwise, it looks great!

**Comment by pnkraemer** [comment]
Thanks! 

(We discussed most of these changes offline, so there is not much to review here.)


So technically, the new behaviour of iclr*() and neurips*() is not backwards-compatible because now they implement times instead of times new roman. 

Not that many would notice a big difference between these two fonts, of course... :)

File: tueplots/fonts.py (hunk 1 above)
Comment (libbylbaker) [reply tueplots/fonts.py]: Okay! Just pushed the changes :)

**Review by pnkraemer — APPROVED** [review]
Thank you for taking care of this!

**Pull request merged**